tests/config/test_search.py

```diff
@@ -9,119 +9,170 @@
+        self.expected = expected
+
+    def check(self, full_path):
+        """Method to match paths that we want"""
+        return full_path in self.expected
 
 
 def test_config_path_no_override():
     """Test we only get our paths"""
-    assert search.config_search_paths() == base_search_paths()
+    assert list(search.config_search_paths()) == list(search.DEFAULT_CONFIG_PATHS)
 
 
 def test_cofig_path_override():
     """Test if a single override is added"""
-    expected = ['/opt']
-    expected.extend(base_search_paths())
-    assert search.config_search_paths('/opt') == expected
+    expected = list(search.DEFAULT_CONFIG_PATHS)
+    expected.append('/opt')
+    assert list(search.config_search_paths('/opt')) == expected
 
 
 def test_config_path_override_list():
     """Test if a list of overrides is added"""
-    expected = ['/opt', '/usr/share']
-    expected.extend(base_search_paths())
-    assert search.config_search_paths(['/opt', '/usr/share']) == expected
+    expected = list(search.DEFAULT_CONFIG_PATHS)
+    expected.extend(['/opt', '/usr/share'])
+    assert list(search.config_search_paths(['/opt', '/usr/share'])) == expected
 
 
 def test_config_path_with_user():
     """Test if a list that contains a path with ~ is expanded"""
-    expected = [os.path.join(os.environ['HOME'], 'my_configs')]
-    expected.extend(base_search_paths())
-    assert search.config_search_paths('~/my_configs') == expected
+    expected = list(search.DEFAULT_CONFIG_PATHS)
+    expected.extend([os.path.join(os.environ['HOME'], 'my_configs')])
+    assert list(search.config_search_paths('~/my_configs')) == expected
 
 
 def test_conf_path_list_user():
     """Test override list has a path with ~"""
-    expected = [os.path.join(os.environ['HOME'], 'my_configs')]
-    expected.extend(base_search_paths())
-    assert search.config_search_paths(['~/my_configs']) == expected
+    expected = list(search.DEFAULT_CONFIG_PATHS)
+    expected.extend([os.path.join(os.environ['HOME'], 'my_configs')])
+    assert list(search.config_search_paths(['~/my_configs'])) == expected
 
 
 def test_default_returned_slot_1():
     """Test config returned from ~/.atkinson"""
     with patch('atkinson.config.search._check_available') as aval_mock:
         aval_mock.return_value = True
-        assert search.get_config_file() == os.path.expanduser('~/.atkinson/config.yml')
+        expected = [os.path.join(x, 'config.yml') for x in search.DEFAULT_CONFIG_PATHS]
+        assert list(search.get_config_files()) == expected
 
 
 def test_default_returned_slot_2():
     """Test config returned from /etc/atkinson"""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [False, True]
-        assert search.get_config_file() == '/etc/atkinson/config.yml'
+        expected = ['/etc/atkinson/config.yml']
+        check = CheckAvailMock(expected)
+        side_effects = check.check
+        aval_mock.side_effect = side_effects
```

Comment:
Maybe collapse these down to:
`aval_mock.side_effect = check.check`
?

Reply:
Bad refactor